test/shortestpaths/dijkstra.jl

```diff
@@ -100,4 +100,171 @@
+
+      for i in 1:5
+        state = dijkstra_shortest_paths(g,i;allpaths=true);
+        for j in 1:5
+          if z.parents[i,j]!=0
+            @test z.parents[i,j] in state.predecessors[j]
+          else
+            @test length(state.predecessors[j]) == 0
+          end
+        end
+      end
+
+      z  = @inferred(floyd_warshall_shortest_paths(g))
+      zm = @inferred(multisource_dijkstra_shortest_paths(g,[1,2]))
+      @test all(isapprox(z.dists[1:2,:],zm.dists))
+      for i in 1:2
+        state = dijkstra_shortest_paths(g,i;allpaths=true);
+        for j in 1:5
+          if z.parents[i,j]!=0
+            @test z.parents[i,j] in state.predecessors[j]
+          else
+            @test length(state.predecessors[j]) == 0
+          end
+        end
+      end
+    end
+
+
+    #Testing multisource On directed Graph
+    g3 = PathDiGraph(5)
+    d = float([ 0 1 2 3 4; 5 0 6 7 8; 9 10 0 11 12; 13 14 15 0 16; 17 18 19 20 0])
+
+    for g in testdigraphs(g3)
+      z  = @inferred(floyd_warshall_shortest_paths(g, d))
+      zm = @inferred(multisource_dijkstra_shortest_paths(g, collect(1:5), d))
+      @test all(isapprox(z.dists,zm.dists))
+
+      for i in 1:5
+        state = dijkstra_shortest_paths(g,i;allpaths=true);
+        for j in 1:5
+          if z.parents[i,j]!=0
+            @test z.parents[i,j] in state.predecessors[j]
+          else
+            @test length(state.predecessors[j]) == 0
+          end
+        end
+      end
+
+      z  = @inferred(floyd_warshall_shortest_paths(g))
+      zm = @inferred(multisource_dijkstra_shortest_paths(g))
+      @test all(isapprox(z.dists,zm.dists))
+
+      for i in 1:5
+        state = dijkstra_shortest_paths(g,i;allpaths=true);
+        for j in 1:5
+          if z.parents[i,j]!=0
+            @test z.parents[i,j] in state.predecessors[j]
+          else
+            @test length(state.predecessors[j]) == 0
+          end
+        end
+      end
+
+      z  = @inferred(floyd_warshall_shortest_paths(g))
+      zm = @inferred(multisource_dijkstra_shortest_paths(g;parallel=true))
+      @test all(isapprox(z.dists,zm.dists))
+
+      for i in 1:5
+        state = dijkstra_shortest_paths(g,i;allpaths=true);
+        for j in 1:5
+          if z.parents[i,j]!=0
+            @test z.parents[i,j] in state.predecessors[j]
+          else
+            @test length(state.predecessors[j]) == 0
+          end
+        end
+      end
+
+      z  = @inferred(floyd_warshall_shortest_paths(g))
+      zm = @inferred(multisource_dijkstra_shortest_paths(g,[1,2]))
+      @test all(isapprox(z.dists[1:2,:],zm.dists))
+      for i in 1:2
+        state = dijkstra_shortest_paths(g,i;allpaths=true);
+        for j in 1:5
+          if z.parents[i,j]!=0
+            @test z.parents[i,j] in state.predecessors[j]
+          else
+            @test length(state.predecessors[j]) == 0
+          end
+        end
+      end
+    end
+
+
+
+    #Testing on 100 vertices and 500 vertices random graphs
```

Comment:
Too big. use a `(10, 50)`. We don't need huge loops in tests. Also, add self loops explicitly if you want them.